### ISSUE TITLE
[Feature Request] An "Attuned Weapon" Weapon Modification to keep track of attuned weapons.

### ISSUE THREAD

**Issue by EmmjayXII** [chummer5a/chummer5a#5078]
Although it is possible to manually keep track of attuned equipment for Adepts/Mystic Adepts, it would be preferable in my opinion to have an "Attuned Item" weapon modification available for all weapons to do so automatically. I am not sure if it would be possible for these attachments to reference the character's initiate grade, but if not, it would be possible to implement as a weapon modification that takes up no slots, is available for all weapons, and has an arbitrarily high maximum rating, adding +1 Accuracy per rating.

If it is possible to reference the character's initiate grade, then it could simply reference that and add that number to Accuracy.

**Comment by Teksura** [comment]
> If it is possible to reference the character's initiate grade, then it could simply reference that and add that number to Accuracy.

My concern with that is while it's doable, you'll have to remove and re-add it when you gain an initiate grade.

**Comment by Teksura** [comment]
Got almost working.

```xml
    <accessory>
      <id>0fcf55fa-afb4-487c-93f2-f41d6fc4e769</id>
      <name>Attuned Item</name>
      <mount />
      <avail>0</avail>
      <cost>0</cost>
      <source>SG</source>
      <page>124</page>
      <rating>0</rating>
      <accuracy>1</accuracy>
      <required>
        <oneof>
          <spell>Attune Item</spell>
        </oneof>
      </required>
    </accessory>
```

This *technically* resolves this, but you're required to add the weapon modification a number of times equal to your IG. Obviously that's not desired. I'm looking into seeing what I have to do to make the bonus pull your IG like how `{MAG}` would pull your magic. If that doesn't work, uhh, can weapon mods even take ratings? I've never tried, and the fact that Gas-Vent system is designed the way it is is... Not encouraging.

**Comment by Teksura** [comment]
So this works fine.... ish.


```xml

    <!-- Region Attune Item -->
    <accessory>
      <id>0fcf55fa-afb4-487c-93f2-f41d6fc4e769</id>
      <name>Attuned Item: (IG 1)</name>
      <mount />
      <avail>0</avail>
      <cost>0</cost>
      <source>SG</source>
      <page>124</page>
      <accuracy>1</accuracy>
      <required>
        <allof>
          <spell>Attune Item</spell>
          <initiategrade>1</initiategrade>
        </allof>
      </required>
      <forbidden>
        <oneof>
          <initiategrade>2</initiategrade>
        </oneof>
      </forbidden>
    </accessory>
    <accessory>
      <id>53286bfd-2d55-4bfa-b7d4-6c55d1d2c7bf</id>
      <name>Attuned Item: (IG 2)</name>
      <mount />
      <avail>0</avail>
      <cost>0</cost>
      <source>SG</source>
      <page>124</page>
      <accuracy>2</accuracy>
      <required>
        <allof>
          <spell>Attune Item</spell>
          <initiategrade>2</initiategrade>
        </allof>
      </required>
      <forbidden>
        <oneof>
          <initiategrade>3</initiategrade>
        </oneof>
      </forbidden>
    </accessory>
```

It's an obnoxious way to go about it and would require a modification for *each* IG, but it works. It's just really jank. I'm going to hold off on pushing this until I hear back on if there is a better way to do the bonus.

**Comment by Teksura** [comment]
> I created it for level 1,2 and 3 => you can mix and have every number in grasp without a dozen of attachments

At that point if you're IG 7 you need to add 3 different attachments as opposed to having just 1.

**Comment by Stexinator** [comment]
> > I created it for level 1,2 and 3 => you can mix and have every number in grasp without a dozen of attachments
> 
> At that point if you're IG 7 you need to add 3 different attachments as opposed to having just 1.

Yes, but how many of your players have IG 7? I agree that this is just a band aid due to not being able to dynamically calculate it. But I made the experience that these 3 are mostly enough

**Comment by Teksura** [comment]
If you don't like the #5084 proposed fix, I invite you to open another pull request to address the issue in a better way.

**Comment by Stexinator** [comment]
Ah, didnt saw that you already provided a fix. I thought we are discussing possible solutions for this issue. I am totally fine with you adding any number :) I just didnt want to do it myself in my custom files
